### ISSUE TITLE
Make error message about too few training examples more informative

### ISSUE THREAD

**Issue by kaiogu** [RasaHQ/rasa#7379]
**Description of Problem**:

The message about too few **core** training examples doesn't inform of what kind of training example is needed (i.e. more intent phrasings or more stories containing the intent). It also doesn't help that the message appears right after `Training NLU model...` giving the impression that this is about nlu data.

```
UserWarning: Intent 'out_of_scope' has only 1 training examples! Minimum is 2, training may fail.
```
**Overview of the Solution**:

One way to make this better would be to explicit what kind of training example is meant:
```
UserWarning: Intent '{intent}' is present in only one story! Minimum is 2, training may fail.
```

Alternatively, raise the warning only after `Training Core model...` which would also help put the warning in context.

**Comment by kaiogu** [comment]
So I tried to reproduce this on a fresh rasa init and didn't manage at first. It doesn't matter if you reduce either Core or NLU examples to 1.

What has to happen here is that there has to be either 1 Core example and 0 NLU or 1 NLU examples and no Core examples.

If you remove both completely (but keep them in your domain) the warning is gone again.

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @Ghostvv will get back to you about it soon! ✨

**Comment by Ghostvv** [comment]
this message comes from nlu, it means that you have only one nlu example for intent `out_of_scope`. Core stories don't influence it

**Comment by kaiogu** [comment]
@Ghostvv 
To reproduce:

1. fresh `rasa init`
2. remove all but one examples from a particular intent in data/nlu.yml
3. `rasa train`-> no warning raised
4. remove all usages of the intent in data/stories.yml
5. `rasa train` -> gives the warning

**Comment by joejuzl** [comment]
I think the problem here is that we are using `TrainingData.training_examples` to determine the number of examples per intent (`number_of_examples_per_intent`). 

However `training_examples` also includes data from `core` - this is where the inconsistent behaviour comes from. 
Having at least one story with an intent increases the number by one, thus often pushing it over the min of 1 when running `rasa train` as opposed to `rasa train nlu`. 
Likewise an intent used in a rule (but not in nlu data) will only show this warning if `rasa train` is used and the intent is loaded.

A solution would be to use `self.nlu_examples` for the count, which filters out core messages. However this would mean we don't get the warning for intents that are used in a rule but not existing in `nlu.yml`.

@Ghostvv thoughts?

**Comment by Ghostvv** [comment]
> Having at least one story with an intent increases the number by one,

why? it shouldn't affect number of examples for this intent, the examples with only intent label (without text) is not used in training of the classifier for this intent

**Comment by joejuzl** [comment]
It comes from `E2EImporter._additional_training_data_from_stories`. Although I am not entirely sure why.

**Comment by ancalita** [comment]
Hi @Ghostvv I discussed with Joe that a possible solution would be to replace `number_of_examples_per_intent `in `TrainingData.validate` method with a different function that counts only nlu training examples. Would this create any inconsistencies with other parts of the code base that uses `number_of_examples_per_intent`?

**Comment by Ghostvv** [comment]
I believe `number_of_examples_per_intent` should only take into account `nlu_examples`.

> Would this create any inconsistencies with other parts of the code base that uses `number_of_examples_per_intent`?

I don't know, needs checking, but the current implementation is incorrect, so its current usage needs to be checked

**Comment by wochinge** [comment]
If I do `Find usages` in Pycharm it show me the following usages of `number_of_examples_per_intent`:
* `print_stats` -> Fine
* `split_nlu_examples` -> Seems your fix would also fix this as the method name already suggests "split **NLU** examples"
* `validate` -> That's the one you want to fix
* `test_train_test_split` -> A test for `split_nlu_examples` which would need adapting

**Comment by Ghostvv** [comment]
`test_train_test_split` should also split only `nlu_examples`

**Comment by ancalita** [comment]
@wochinge are you suggesting I should create a test for `split_nlu_examples`? There's not currently one. Otherwise, `test_train_test_split `passes after the change to `nlu_examples` in `number_of_examples_per_intent`. I also created a test for `number_of_examples_per_intent `that passes (and fails before the change).

**Comment by wochinge** [comment]
@ancalita Isn't `test_train_test_split` covering `split_nlu_examples`?

**Comment by ancalita** [comment]
@wochinge Yep, you're right, that's covered in `test_train_test_split`. What is slightly strange is that there's no test for `validate` - should probably add one at least for the scenario involving `number_of_examples_per_intent` 😄 ?

**Comment by wochinge** [comment]
>  should probably add one at least for the scenario involving number_of_examples_per_intent 😄

That would be dope! 💯
